flink-runtime/src/main/java/org/apache/flink/runtime/state/CheckpointStorageLoader.java

```diff
@@ -175,21 +174,16 @@ public static CheckpointStorage load(
                     CheckpointingOptions.SAVEPOINT_DIRECTORY, defaultSavepointDirectory.toString());
         }
 
-        // Legacy state backends always take precedence for backwards compatibility.
-        StateBackend rootStateBackend =
-                (configuredStateBackend instanceof DelegatingStateBackend)
-                        ? ((DelegatingStateBackend) configuredStateBackend)
-                                .getDelegatedStateBackend()
-                        : configuredStateBackend;
-
-        if (rootStateBackend instanceof CheckpointStorage) {
```

Comment:
Could you please explain the removal? 
IIUC, changelog state backend can still be passed here (even if hidden from user) when called from `StreamTask#createCheckpointStorage`

cc: @curcur

Reply:
You are right. This should not be removed. I made a mistake.
I revert this change. Thanks a lot for pointing this out.